check_api/src/main/java/com/google/errorprone/ErrorPronePlugins.java

```diff
@@ -45,11 +54,50 @@ public static ScannerSupplier loadPlugins(ScannerSupplier scannerSupplier, Conte
       return scannerSupplier;
     }
     ClassLoader loader = fileManager.getClassLoader(StandardLocation.ANNOTATION_PROCESSOR_PATH);
-    Iterable<BugChecker> extraBugCheckers = ServiceLoader.load(BugChecker.class, loader);
-    if (Iterables.isEmpty(extraBugCheckers)) {
+
+    List<Class<? extends BugChecker>> bugCheckerClasses;
+    // Synchronize to prevent races
+    synchronized(ErrorPronePlugins.class) {
+      // disable URL caching to avoid stale state bugs when run within a daemon-building process
```

Comment:
@cushon I don't think there is anything specifically wrong with how EP uses service loading.  Any build tool with a daemon that runs javac in-process could run into problems.  We were trying to fix the problem we were observing in the narrowest way possible, which is hard given the weird global setting.  A workaround at the Buck level might have other unanticipated effects, but I'll think about alternate solutions a bit more.

BTW, could Bazel builds run into this same problem?  I think Bazel also has a daemon.